src/go/internal/testutil/fs.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2021 The Magma Authors.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutil
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+// MustTempDir returns a randomly generated temp dir from ioutil.TempDir and
+// a cleanup function. If the test failed, the temp dir is not cleaned up to
+// aid in debugging. Any encountered err results in a panic.
```

Comment:
should we clean temp directories before the next run?
maybe add prefix so that we can find them

Reply:
we can't really safely clean the system-wide temp dir (e.g. magma uses /tmp for sctpd <-> mme unix domain sockets). Also, CI envs should start clean on each run in an ideal world :)

Reply:
Have you previously used this paradigm Mark? Like Eli I do worry about tmp refuse building up (but I also see the strong value of seeing what went wrong after a test fails).  What if we default to tmpdir cleanup but a flag allows preservation for debug? It sounds like nominal workflow is to clean up (both per ioutil.TempDir Godocs and for example when I explore [Bazel's recommended handling of TempDirs](https://docs.bazel.build/versions/main/test-encyclopedia.html#test-interaction-with-the-filesystem).

Reply:
Good points; I'll remove this "enhancement". Tests should be repeatable anyways, so folks should be able to re-run anywhere to debug failures